presto-spark-base/src/main/java/com/facebook/presto/spark/execution/nativeprocess/HttpNativeExecutionTaskInfoFetcher.java

```diff
@@ -91,55 +91,8 @@ public HttpNativeExecutionTaskInfoFetcher(
 
     public void start()
     {
-        scheduledFuture = updateScheduledExecutor.scheduleWithFixedDelay(() ->
-        {
-            try {
-                ListenableFuture<BaseResponse<TaskInfo>> taskInfoFuture = workerClient.getTaskInfo();
-                Futures.addCallback(
-                        taskInfoFuture,
-                        new FutureCallback<BaseResponse<TaskInfo>>()
-                        {
-                            @Override
-                            public void onSuccess(BaseResponse<TaskInfo> result)
-                            {
-                                log.debug("TaskInfoCallback success %s", result.getValue().getTaskId());
-                                taskInfo.set(result.getValue());
-                                if (result.getValue().getTaskStatus().getState().isDone()) {
-                                    synchronized (taskFinished) {
-                                        taskFinished.notifyAll();
-                                    }
-                                }
-                            }
-
-                            @Override
-                            public void onFailure(Throwable t)
-                            {
-                                // record failure
-                                try {
-                                    errorTracker.requestFailed(t);
-                                }
-                                catch (PrestoException e) {
-                                    stop();
-                                    lastException.set(e);
-                                    return;
-                                }
-                                ListenableFuture<?> errorRateLimit = errorTracker.acquireRequestPermit();
-                                try {
-                                    // synchronously wait on throttling
-                                    errorRateLimit.get(maxErrorDuration.toMillis(), TimeUnit.MILLISECONDS);
-                                }
-                                catch (InterruptedException | ExecutionException | TimeoutException e) {
-                                    // throttling error is not fatal, just log the error.
-                                    log.debug(e.getMessage());
-                                }
-                            }
-                        },
-                        executor);
-            }
-            catch (Throwable t) {
-                throw t;
-            }
-        }, 0, (long) infoFetchInterval.getValue(), infoFetchInterval.getUnit());
+        scheduledFuture = updateScheduledExecutor.scheduleWithFixedDelay(
+                this::doGetTaskInfo, 0, (long) infoFetchInterval.getValue(), infoFetchInterval.getUnit());
```

Comment:
@vermapratyush This is an interesting point. I looked into this. I am thinking, instead of enforcing config conpatibility, we could leverage the thread pool scheduler logic (`schedulerAtFixedDelay`) to do this.

The guarantee that this `schedulerAtFixedDelay` offers is that it will not schedule the next instance of the work until - the previous instance returns + the specified delay. 
So cleaning up this callback logic like this might be a good solution ? https://github.com/shrinidhijoshi/presto/pull/1

Reply:
Sounds pretty good to me. The callback seemed unnecessary here.